lib/state/ui/reducer.ts

```diff
@@ -22,4 +26,42 @@ const visiblePanes = (state = defaultVisiblePanes, { type, show }) => {
   return state;
 };
 
-export default combineReducers({ filteredNotes, visiblePanes });
+const setSelectedNote = (state = null, action) => {
+  switch (action.type) {
+    case 'App.notesLoaded': {
+      return state
+        ? action.notes.find(({ id }) => id === state.id) || state
+        : action.notes[0] || null;
+    }
+    case 'App.noteUpdatedRemotely':
+    case 'App.selectNote': {
+      return {
+        ...action.note,
+        hasRemoteUpdate: action.hasRemoteUpdate,
+      };
+    }
+    case 'App.closeNote':
+    case 'App.showAllNotes':
+    case 'App.selectTrash':
+    case 'App.selectTag':
+      return null;
+    case SET_SELECTED_NOTE:
+      return action.note;
+    case 'FILTER_NOTES':
+      if (
+        action.notes.length &&
+        (!state || !action.notes.some(note => note.id === state.id))
```

Comment:
it looks like we'll always have `notes` as an array…this shouldn't need the check…

```ts
import { FILTER_NOTES } from '../action-types'

case FILTER_NOTES:
  return state && action.notes.some(({id}) => id === state.id)
    ? state
    : action.notes[0] || null;
```